text/0111-pure-proxy-replication.md

```diff
@@ -0,0 +1,132 @@
+
+To prove that the pure account is keyless, the client must provide the initial preimage used by the chain to derive the pure account. Parachain `A` verifies it and sends it to parachain `B` with the replication request.
+
+We can draft a pallet extension for the proxy pallet, which needs to be initialised on both sides to enable replication:
+
+``` rust 
+// Simplified version to illustrate the concept.
+mod pallet_proxy_replica {
+  /// The part of the pure account preimage that has to be provided by a client.
+  struct Witness {
+    /// Pure proxy swapner
+    spawner: AccountId,
+    /// Disambiguation index
+    index: u16,
+    /// The block height and extrinsic index of when the pure account was created.  
+    block_number: BlockNumber,
+    /// The extrinsic index.
+    ext_index: u32,
+    // Part of the preimage, but constant.
+    // proxy_type: ProxyType::Any,
+  } 
+  // ...
+  
+  /// The replication call to be initiated on the source chain.
+  // Simplified version, the XCM part will be abstracted by the `Config` trait.
+  fn replicate(origin: SignedOrigin, witness: Witness, proxy: xcm::Location) -> ... {
+       let pure = ensure_signed(origin);
+       ensure!(pure == proxy_pallet::derive_pure_account(witness), Error::NotPureAccount);
+       let xcm = vec![
+         DescendOrigin(who),
+         Transact(
+             // …
+             origin_kind: OriginKind::Xcm,
+	     call: pallet_proxy_replica::create(witness, proxy).encode(),
+         )
+       ];
+       xcmTransport::send(xcm)?;
+  }
+  // …
+  
+  /// The call initiated by the source chain on the receiving chain.
+  // `Config::CreateOrigin` - generally open for whitelisted parachain IDs and 
+  // converts `Origin::Xcm(ParachainA/AccountId32(pure1))` to `AccountID(pure1)`.
+  fn create(origin: Config::CreateOrigin, witness: Witness, proxy: xcm::Location) -> ... {
+       let pure = T::CreateOrigin::ensure_origin(origin);
+       ensure!(pure == proxy_pallet::derive_pure_account(witness), Error::NotPureAccount);
+       proxy_pallet::create_pure_proxy(pure, proxy);
+  }
+}
+
+```
+
+## Drawbacks
+
+There are two disadvantages to this approach:
+- The receiving chain has to trust the sending chain's claim that the account controlling the pure account has commanded the replication.
+- Clients must obtain witness data.
+
+We could eliminate the first disadvantage by allowing only the spawner of the pure proxy to recreate the pure proxies, if they sign the transaction on a remote chain and supply the witness/preimage. Since the preimage of a pure account includes the account ID of the spawner, we can verify that the account signing the transaction is indeed the spawner of the given pure account. However, this approach would grant exclusive rights to the spawner over the pure account, which is not a property of pure proxies at present. This is why it's not an option for us.
```

Comment:
yeah we shouldn't do this. it is for example the spawner private key is forgotten because people wasn't expecting they need it anymore